### PR TITLE
feat: tree-select supports onPopupScroll

### LINKED ISSUE #48528: onPopScroll function not working for TreeSelect
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/p/sandbox/antd-reproduction-template-forked-w8hflc?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clv4xnb3200062v6iswwd3ixn%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clv4xnb3200022v6i0ez9qvlt%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clv4xnb3200032v6iszd3a1gl%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clv4xnb3200052v6i920zu3sk%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clv4xnb3200022v6i0ez9qvlt%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clv4xnb3200012v6i31zmobr4%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Findex.js%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clv4xnb3200022v6i0ez9qvlt%2522%252C%2522activeTabId%2522%253A%2522clv4xnb3200012v6i31zmobr4%2522%257D%252C%2522clv4xnb3200052v6i920zu3sk%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clv4xnb3200042v6ijpn2fwq9%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clv4xnb3200052v6i920zu3sk%2522%252C%2522activeTabId%2522%253A%2522clv4xnb3200042v6ijpn2fwq9%2522%257D%252C%2522clv4xnb3200032v6iszd3a1gl%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clv4xnb3200032v6iszd3a1gl%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

### Steps to reproduce

Just open the treeselect , expand the first node and scroll to bottom.

### What is expected?

The onPopupScroll or onSroll functions should be fired.

### What is actually happening?

The onPopupScroll or onSroll functions are not being fired.

| Environment | Info |
| --- | --- |
| antd | 5.5.1 |
| React | 18.2.0 |
| System | Ubuntu 22.04.4 LTS |
| Browser | Chrome 124.0.6367.60 |

---
If possible please provide a indepth documentation or code implementation

### PR DESCRIPTION
**Pull request by wanpan11**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [x] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close #48528

### 💡 Background and solution

tree-select supports onPopupScroll

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     tree-select supports onPopupScroll      |
| 🇨🇳 Chinese |      tree-select 支持 onPopScroll     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by li-jia-nan** [comment]
新特性提交到 feature 分支吧

File: components/tree-select/index.en-US.md [hunk 1]
```diff
@@ -86,6 +86,7 @@ Common props ref：[Common props](/docs/react/common-props)
 | onSearch | A callback function, can be executed when the search input changes | function(value: string) | - |  |
 | onSelect | A callback function, can be executed when you select a treeNode | function(value, node, extra) | - |  |
 | onTreeExpand | A callback function, can be executed when treeNode expanded | function(expandedKeys) | - |  |
+| onPopupScroll | Called when dropdown scrolls | (event: MouseEvent) => void | - | 5.16.5 |
```
Comment (li-jia-nan) [inline components/tree-select/index.en-US.md:89]:
```suggestion
| onPopupScroll | Called when dropdown scrolls | (event: MouseEvent) => void | - | 5.17.0 |
```

File: components/tree-select/index.zh-CN.md [hunk 2]
```diff
@@ -87,6 +87,7 @@ demo:
 | onSearch | 文本框值变化时的回调 | function(value: string) | - |  |
 | onSelect | 被选中时调用 | function(value, node, extra) | - |  |
 | onTreeExpand | 展示节点时调用 | function(expandedKeys) | - |  |
+| onPopupScroll | 下拉列表滚动时的回调 | (event: MouseEvent) => void | - | 5.16.5 |
```
Comment (li-jia-nan) [inline components/tree-select/index.zh-CN.md:90]:
```suggestion
| onPopupScroll | 下拉列表滚动时的回调 | (event: MouseEvent) => void | - | 5.17.0 |
```

**Comment by li-jia-nan** [comment]
master 的提交带过来了，需要清理一下 commit，只保留自己的改动就好了

**Comment by wanpan11** [comment]
> master 的提交带过来了，需要清理一下 commit，只保留自己的改动就好了

好的

**Comment by wanpan11** [comment]
@li-jia-nan 佬 这个 ci 报错 我需要干撒不

**Comment by li-jia-nan** [comment]
> @li-jia-nan 佬 这个 ci 报错 我需要干撒不

看起来不是这个 pr 导致的，不过得开个新 PR 修一下，感觉是哪个库更新了，求助一下大佬 @zombieJ @MadCcc

**Comment by wanpan11** [comment]
不知道是不是版本的问题，我本地 18.2 是通过了的。
image

**Comment by li-jia-nan** [comment]
我来看一下吧

**Review by MadCcc — DISMISSED** [review]

File: components/tree-select/demo/basic.tsx [hunk 3]
```diff
@@ -51,6 +67,9 @@ const App: React.FC = () => {
       treeDefaultExpandAll
       onChange={onChange}
       treeData={treeData}
+      onPopupScroll={(e) => {
```
Comment (li-jia-nan) [inline components/tree-select/demo/basic.tsx:70]: 上面的 onChange 方法单独拿出去了，感觉这里也可以拿出去，保持一致

Comment (wanpan11) [reply components/tree-select/demo/basic.tsx]: 🆗

**Comment by wanpan11** [comment]
@SocketSecurity ignore npm/@ant-design/tools@18.0.2
@SocketSecurity ignore npm/@biomejs/biome@1.7.1

**Comment by li-jia-nan** [comment]
@SocketSecurity ignore npm/@ant-design/tools@18.0.2
@SocketSecurity ignore npm/@biomejs/biome@1.7.1

**Review by li-jia-nan — APPROVED** [review]

**Pull request merged**